MIGRATIONGUIDE.md

```diff
@@ -0,0 +1,363 @@
+ creation/update, and had built business logic around this assumption, after [version change](https://stripe.com/docs/upgrades#2018-09-06)
+  the validation is removed and SKUs returned can have duplicate attributes. Your code 
+  depending on SKU attribute uniqueness has to be updated. 
+  
+Parameters can also take new default values. Your old code may not explicitly pass the parameter 
+value, but after version upgrade when default value kicks in, your same API call now has a 
+different business implication. For example, in [version change](https://stripe.com/docs/upgrades#2018-05-21), 
+`Subscription#update` will take `trial_from_plan` as true if it is not explicitly specified.
+
+#### API errors
+
+Less obvious ones are changes related to JSON error object which is not directly exposed to you. 
+This library converts such failure into `StripeException` a subclass of Java `Exception`. For 
+example, in [version change](https://stripe.com/docs/upgrades#2017-12-14) `card_error` is 
+introduced, but that will be surfaced to you simply as a different message in `ApiException`. No 
+change on your part is required.
+
+Change in HTTP error codes, however, can affect your integration as a different type of 
+`StripeException` thrown. For example, in [version change](https://stripe.com/docs/upgrades#2016-10-19),
+ insufficient permission will return code 403 instead of 401. This means your code checking for 
+ `AuthenticationException` may require updates to `PermissionException` instead. Similarly, 429 is 
+ introduced after [version change](https://stripe.com/docs/upgrades#2015-09-08) instead of 400. 
+ As a result, any code handling rate-limit should be updated to check for `RateLimitException`.
+
+#### Summary
+ 
+Please consult the API change log for more details. A complementary recommendation for 
+`stripe-java` is to pay close attention for the followings;
+
+* API response (beyond compiler complaints)
+  - enum values represented as String, 
+  - `object` field values renaming
+  - change in business meaning of the field
+* API request
+  - parameter removal
+  - parameter validation and effect on resulting API response
+  - parameter default value
+* API error
+  - HTTP error code and its corresponding `StripeException`
+  
+### 2. Event response structure
+
+The official upgrade [guide](https://stripe.com/docs/upgrades#how-can-i-upgrade-my-api) 
+recommends  configuring one test webhook to the latest API version. This allows you to test 
+handling event and its event data object rendered at the upgraded API version. In the latest 
+`stripe-java`, you will find an API version defined at its static value 
+`Stripe#API_VERSION` matching the latest API version you can configure your test webhook to.
+ 
+#### Motivation for handling event compatibility
+ 
+The official guide also recommends the need for compatibility in handling events of both your 
```

Comment:
`The official guide also recommends the need for compatibility in` -> `We also recommend maintaining compatibility for`

Reply:
Hmm similar to the above, I would like to defer full authority explicitly to the official guide, because this migration should serve as additional explanation -- developers shouldn't only read this guide alone.